gravitee-am-service/src/main/java/io/gravitee/am/service/impl/TokenServiceImpl.java

```diff
@@ -113,7 +123,13 @@ public Completable deleteByUserId(String userId) {
                     LOGGER.error("An error occurs while trying to delete tokens by user {}", userId, ex);
                     return Completable.error(new TechnicalManagementException(
                             String.format("An error occurs while trying to find total tokens by user: %s", userId), ex));
-                });
+                })
+                .doOnEvent(error -> userService.findById(userId).doOnEvent((user, er) ->
```

Comment:
**suggestion**: I don't think we need to do a findUser here as the user is loaded in the caller object of the `deleteByUserId` method. Instead of doing this search which may fail, why not updating the method signature to accept a User object?

Reply:
there is a need to use a user object to set an correct domain